diesel/src/main.rs

```diff
@@ -56,6 +59,59 @@ fn index(
         .responder()
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+struct MyUser {
+    name: String
+}
+
+const MAX_SIZE: usize = 262_144; // max payload size is 256k
+
+/// This handler manually load request payload and parse json object
+fn index_add(req: &HttpRequest<AppState>) -> Box<Future<Item = HttpResponse, Error = Error>> {
```

Comment:
Should you use here `Json` extractor instead of manually parsing `MyUser`?

Reply:
I don't see why I should be using a different extractor type, as this is a situation that should be workable. I have the requirement to add extra data into the error messages based on parsing issues, so I need to use the manual method to create those errors. I will also extend this example to have the extractor type, but I'd really appreciate advice on making this situation work. I've also got a similar example of the same with async that fails in it's own unique way. (I also seem to recall that the extractor type has a very similar set of issues and I can't get that working either ....)

Reply:
As noted, extractor causes the following errors. I'll post the code to this PR now. 
```
error[E0277]: the trait bound `actix_web::HttpResponse: futures::Future` is not satisfied
   --> diesel/src/main.rs:125:10
    |
125 |         .and_then(|res| match res {
    |          ^^^^^^^^ the trait `futures::Future` is not implemented for `actix_web::HttpResponse`
    |
    = note: required because of the requirements on the impl of `futures::IntoFuture` for `actix_web::HttpResponse`

error[E0308]: mismatched types
   --> diesel/src/main.rs:119:5
    |
118 |   fn extract_item_limit((item, req): (Json<MyUser>, HttpRequest<AppState>)) -> HttpResponse {
    |                                                                                ------------ expected `actix_web::HttpResponse` because of return type
119 | /     req.state()
120 | |         .db
121 | |         .send(CreateUser {
122 | |             name: item.name,
...   |
127 | |             Err(_) => HttpResponse::InternalServerError().into(),
128 | |         })
    | |__________^ expected struct `actix_web::HttpResponse`, found struct `futures::AndThen`
    |
    = note: expected type `actix_web::HttpResponse`
               found type `futures::AndThen<futures::future::FromErr<actix::prelude::Request<db::DbExecutor, db::CreateUser>, _>, actix_web::HttpResponse, [closure@diesel/src/main.rs:125:19: 128:10]>`

error[E0282]: type annotations needed
   --> diesel/src/main.rs:153:55
    |
153 |                     .with_config(extract_item_limit, |((cfg, _),)| {
    |                                                       ^^^^^^^^^^^ consider giving this closure parameter a type
    |
    = note: type must be known at this point

error: aborting due to 4 previous errors

```

Reply:
I suspect there might be ambiguous type in `with_config` because you failed to compile `extract_item_limit`

Reply:
Hey there, that seems to be separate from the issue. I have found no way to make cfg work with the with_config type and a future response.